src/wasm.rs

```diff
@@ -207,6 +266,44 @@ mod test {
         assert!(t.is_some());
     }
 
+    #[test]
+    fn decode_wasi_preview() {
+        let mut f = File::open("./fixtures/guest.component.wasm").unwrap();
```

Comment:
this should be the preview path right?

Reply:
You mentioned that the wasi_preview_snapshot.wasm file was just a "regular" old type file, so I didn't think it was worth testing since we already have the other fixture. Maybe the test here is poorly named, as it's an encode/decode test on the component-embedded file.

Reply:
Got it, ty